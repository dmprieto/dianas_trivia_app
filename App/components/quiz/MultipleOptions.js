import { Pressable, Text, View } from "react-native"

export const MultipleOptions = ({ options, answer, setAnswer }) => {
  const optionStyles = "bg-sky-500 py-2 px-4 rounded ml-4 mt-4"

  return (
    <View className="flex-auto">
      {options.map((option) => {
        let opacity = ""
        if (answer !== "" && answer === option.name) {
          opacity += "opacity-1"
        } else {
          opacity += "opacity-70"
        }
        return (
          <Pressable
            className={`${optionStyles} ${opacity}`}
            key={option.id}
            onPress={() => {
              setAnswer(option.name)
            }}
          >
            <Text className="text-xl text-white font-bold">{option.name}</Text>
          </Pressable>
        )
      })}
    </View>
  )
}
