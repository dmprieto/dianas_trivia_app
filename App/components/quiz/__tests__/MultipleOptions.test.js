import renderer from "react-test-renderer"

import { MultipleOptions } from "../MultipleOptions"

describe("<MultipleOptions />", () => {
  const options = [{ id: "opt1", name: "" }]
  const answer = ""
  const setAnswer = () => {}
  it("has 1 child", () => {
    const tree = renderer
      .create(
        <MultipleOptions
          options={options}
          answer={answer}
          setAnswer={setAnswer}
        />
      )
      .toJSON()
    expect(tree.children.length).toBe(1)
  })

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MultipleOptions
          options={options}
          answer={answer}
          setAnswer={setAnswer}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
