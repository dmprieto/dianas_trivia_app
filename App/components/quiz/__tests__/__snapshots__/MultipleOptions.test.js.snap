// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultipleOptions /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexShrink": 1,
      },
      Object {
        "width": 240,
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "opt1",
          "name": "",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#0ea5e9",
              },
              Object {
                "paddingBottom": 8,
                "paddingTop": 8,
              },
              Object {
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "borderBottomLeftRadius": 4,
                "borderBottomRightRadius": 4,
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              Object {
                "marginTop": 16,
              },
              Object {
                "opacity": 0.7,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontWeight": "700",
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
